numpy/core/getlimits.py

```diff
@@ -208,23 +277,26 @@ def _register_known_types():
     # https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#Double-double_arithmetic
     # These numbers have the same exponent range as float64, but extended number of
     # digits in the significand.
-    huge_dd = (umath.nextafter(ld(inf), ld(0))
-                if hasattr(umath, 'nextafter')  # Missing on some platforms?
-                else float64_ma.huge)
+    huge_dd = nextafter(ld(inf), ld(0), dtype=ld)
+    # As the smallest_normal in double double is so hard to calculate we set
+    # it to NaN.
+    smallest_normal_dd = NaN
     float_dd_ma = MachArLike(ld,
-                              machep=-105,
-                              negep=-106,
-                              minexp=-1022,
-                              maxexp=1024,
-                              it=105,
-                              iexp=11,
-                              ibeta=2,
-                              irnd=5,
-                              ngrd=0,
-                              eps=exp2(ld(-105)),
-                              epsneg= exp2(ld(-106)),
-                              huge=huge_dd,
-                              tiny=exp2(ld(-1022)))
+                             machep=-105,
+                             negep=-106,
+                             minexp=-1022,
+                             maxexp=1024,
+                             it=105,
+                             iexp=11,
+                             ibeta=2,
+                             irnd=5,
+                             ngrd=0,
+                             eps=exp2(ld(-105)),
+                             epsneg=exp2(ld(-106)),
+                             huge=huge_dd,
+                             tiny=smallest_normal_dd,
```

Comment:
Do we consider `MachArLike` private?  In that case we could just rename `tiny` to `smallest_normal` here.  Otherwise, maybe skip `smallest_normal` and set it in the init to avoid the duplication?

Reply:
I'm not sure how we want to perform the transition here, for backwards compatibility we will have those two duplicated inside the class but I don't know what keyword in the constructor should we leave for the long term. I can leave in the constructor only `smallest_normal`

Reply:
I dunno, even JAX which has `bfloat16` and [subclasses `np.finfo`](https://github.com/google/jax/pull/5486/files) does not use it:
```
obj.machar = None  # np.core.getlimits.MachArLike does not support bfloat16.
```
(I honestly don't think this should be public to begin with, but that is a follow-up.)


But well, we can try to keep it compatible for now (which I guess means passing *only* `tiny`)?  And should probably deprecate the whole thing (i.e. make it non-public for good).  It shouldn't be public, but maybe we have to go the slow route.